### PR TITLE
Changed Registration from type alias to case class

### PR DESCRIPTION
**Pull request by MarconZet**

### CODE REVIEW

**Review by PawelLipski — APPROVED** [review]
_Dobra zmiana_ :pis:

File: circe-akka-serializer/src/main/scala/org/virtuslab/ash/circe/CirceAkkaSerializer.scala [hunk 1]
```diff
@@ -163,7 +165,7 @@ abstract class CirceAkkaSerializer[Ser <: AnyRef: ClassTag](system: ExtendedActo
    */
   protected def genericCodec: Codec[Ser] = Codec.from(genericDecoder, genericEncoder)
 
-  protected def genericEncoder: Encoder[Ser] =
+  private def genericEncoder: Encoder[Ser] =
```
Comment (PawelLipski) [inline circe-akka-serializer/src/main/scala/org/virtuslab/ash/circe/CirceAkkaSerializer.scala:168]: This visibility change is unrelated to the tuple=>case class change, right? 🤔

Comment (PawelLipski) [reply circe-akka-serializer/src/main/scala/org/virtuslab/ash/circe/CirceAkkaSerializer.scala:168]: Isn't it more related to #98 (?)

Comment (MarconZet) [reply circe-akka-serializer/src/main/scala/org/virtuslab/ash/circe/CirceAkkaSerializer.scala:168]: It's a refactor I wanted to do and decided to do it now, because it is not backward compatible.

Comment (PawelLipski) [reply circe-akka-serializer/src/main/scala/org/virtuslab/ash/circe/CirceAkkaSerializer.scala:168]: Btw if you could open a PR to Hydra which adapts to all the new changes (probably as v0.3.3 then), after doing #98
